src/twisted/web/http.py

```diff
@@ -1825,75 +1823,123 @@ class _ChunkedTransferDecoder:
 
     state = "CHUNK_LENGTH"
 
-    def __init__(self, dataCallback, finishCallback):
+    def __init__(
+        self,
+        dataCallback: Callable[[bytes], None],
+        finishCallback: Callable[[bytes], None],
+    ) -> None:
         self.dataCallback = dataCallback
         self.finishCallback = finishCallback
-        self._buffer = b""
+        self._buffer = bytearray()
+        self._start = 0
 
-    def _dataReceived_CHUNK_LENGTH(self, data):
-        if b"\r\n" in data:
-            line, rest = data.split(b"\r\n", 1)
-            parts = line.split(b";")
-            try:
-                self.length = int(parts[0], 16)
-            except ValueError:
-                raise _MalformedChunkedDataError("Chunk-size must be an integer.")
-            if self.length < 0:
-                raise _MalformedChunkedDataError("Chunk-size must not be negative.")
-            elif self.length == 0:
-                self.state = "TRAILER"
-            else:
-                self.state = "BODY"
-            return rest
+    def _dataReceived_CHUNK_LENGTH(self) -> bool:
+        """
+        Read the chunk size line, ignoring any extensions.
+
+        @returns: C{True} once the line has been read and removed from
+            C{self._buffer}.  C{False} when more data is required.
+
+        @raises _MalformedChunkedDataError: when the chunk size cannot be
+            decoded.
+        """
+        eolIndex = self._buffer.find(b"\r\n", self._start)
+        if eolIndex == -1:
+            self._start = len(self._buffer) - 1
+            return False
+
+        endOfLengthIndex = self._buffer.find(b";", 0, eolIndex)
+        if endOfLengthIndex == -1:
+            endOfLengthIndex = eolIndex
+        try:
+            length = int(self._buffer[0:endOfLengthIndex], 16)
+        except ValueError:
+            raise _MalformedChunkedDataError("Chunk-size must be an integer.")
+
+        if length < 0:
+            raise _MalformedChunkedDataError("Chunk-size must not be negative.")
+        elif length == 0:
+            self.state = "TRAILER"
         else:
-            self._buffer = data
-            return b""
+            self.state = "BODY"
 
-    def _dataReceived_CRLF(self, data):
-        if data.startswith(b"\r\n"):
+        self.length = length
+        del self._buffer[0 : eolIndex + 2]
+        self._start = 0
+        return True
+
+    def _dataReceived_CRLF(self) -> bool:
+        """
+        Await the carriage return and line feed characters that follow the
+        chunk data.
+
+        @returns: C{True} when the CRLF have been read, otherwise C{False}.
+        """
+        # TODO: https://twistedmatrix.com/trac/ticket/10137
```

Comment:
thanks for the todo.
When adding a TODO/FIXME i prefer to also leave a short comment.
In this way, I don't have to check Trac each time I see a todo.
Just a short to help you decide if you should check the ticket or ignore it

```suggestion
        # TODO: https://twistedmatrix.com/trac/ticket/10137
        # It should raise an error when receiving malformed end of chunk markers.
```

Reply:
Agreed, but I intend to fix this in the next few days so I didn't bother.